src/ApiService/ApiService/onefuzzlib/NsgOperations.cs

```diff
@@ -97,11 +97,17 @@ await _creds.GetResourceGroupResource()
         }
 
         public async Async.Task<NetworkSecurityGroupResource?> GetNsg(string name) {
-            var response = await _creds.GetResourceGroupResource().GetNetworkSecurityGroupAsync(name);
-            if (response == null) {
-                //_logTracer.Debug($"nsg %s does not exist: {name}");
+            try {
+                var response = await _creds.GetResourceGroupResource().GetNetworkSecurityGroupAsync(name);
+                return response?.Value;
+            } catch (RequestFailedException ex) {
```

Comment:
Can `GetNetworkSecurityGroupAsync` throw `RequestFailedException`? I don't see it in the docs but I also assumed (maybe incorrectly) that those docs list all the exceptions that could be thrown.

Docs: https://docs.microsoft.com/en-us/dotnet/api/azure.resourcemanager.network.networkextensions.getnetworksecuritygroupasync?view=azure-dotnet-preview#exceptions

Reply:
```
{"The Resource 'Microsoft.Network/networkSecurityGroups/anNsg' under resource group '1f-daily' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix\r\nStatus: 404 (Not Found)\r\nErrorCode: ResourceNotFound\r\n\r\nContent:\r\n{\"error\":{\"code\":\"ResourceNotFound\",\"message\":\"The Resource 'Microsoft.Network/networkSecurityGroups/anNsg' under resource group  was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix\"}}\r\n\r\nHeaders:\r\nCache-Control: no-cache\r\nPragma: no-cache\r\nx-ms-failure-cause: REDACTED\r\nx-ms-request-id: 30699c4f-585a-42d5-a491-79b8007faf64\r\nx-ms-correlation-request-id: REDACTED\r\nx-ms-routing-request-id: REDACTED\r\nStrict-Transport-Security: REDACTED\r\nX-Content-Type-Options: REDACTED\r\nDate: Fri, 06 May 2022 19:03:51 GMT\r\nContent-Type: application/json; charset=utf-8\r\nExpires: -1\r\nContent-Length: 225\r\n"}
```

It does, I actual discover this by running the code and testing